py-dipy: Update version to support python@3.10

payerle opened the pull request:
py-dipy only adds support to python@3.10 in py-dipy@1.5.0

Closes #40228

payerle committed:
py-dipy: Update version to support python@3.10

py-dipy only adds support to python@3.10 in py-dipy@1.5.0

See #40228

payerle committed:
py-dipy: fix formatting issues

payerle committed:
py-dipy: another formatting fix

payerle committed:
py-dipy: Use depends instead of conflicts

payerle committed:
py-dipy: formatting fix

manuelakuhn commented on var/spack/repos/builtin/packages/py-dipy/package.py:
```diff
@@ -19,9 +19,11 @@ class PyDipy(PythonPackage):
     version("1.4.1", sha256="b4bf830feae751f3f985d54cb71031fc35cea612838320f1f74246692b8a3cc0")
 
     depends_on("python@3.6:", type=("build", "run"))
+    depends_on("python@:3.9", type=("build", "run"), when="@:1.4")
```
optional:
switch the order of the `type` and `when` argument as most of the other python packages do

manuelakuhn reviewed (commented):
The minimum version requirements changed for 1.7.0. Can you add them?

CYTHON_MIN_VERSION = '0.29.24'
NUMPY_MIN_VERSION = '1.14.5'
SCIPY_MIN_VERSION = '1.1'
NIBABEL_MIN_VERSION = '3.0.0'
H5PY_MIN_VERSION = '2.8.0'
(see [dipy/info.py](https://github.com/dipy/dipy/blob/1.7.0/dipy/info.py#L77C1-L81C27))

payerle committed:
py-dipy: Updating for @1.7.0

Added new minimum version requirements for
py-cython
py-numpy
py-scipy
py-h5py
as suggested by @manuelakuhn
(py-nibabel min version unchanged for @1.7.0)

manuelakuhn reviewed (approved):
If you want to be exact, the new restricted dependencies are present starting from `py-dipy@1.5:`
```suggestion
    depends_on("py-cython@0.29.24:", type=("build"), when="@1.5:")
    depends_on("py-numpy@1.14.5:", type=("build", "run"), when="@1.5:")
    depends_on("py-scipy@1.1:", type=("build", "run"), when="@1.5:")
    depends_on("py-h5py@2.8.0:", type=("build", "run"), when="@1.5:")
```

adamjstewart reviewed (approved):
Good enough
